Library/ViewModels/DiscoveryFiltersViewModel.swift

```diff
@@ -59,28 +65,55 @@ public final class DiscoveryFiltersViewModel: DiscoveryFiltersViewModelType,
       .take(first: 1)
       .map { topFilters(forUser: AppEnvironment.current.currentUser) }
 
-    let initialSelectedRowWithCategories = Signal.combineLatest(
-      self.initialSelectedRowWithCategoriesProperty.signal.skipNil(),
+    let initialSelectedRow = Signal.combineLatest(
+      self.initialSelectedRowProperty.signal.skipNil(),
       self.viewDidLoadProperty.signal
       )
       .take(first: 1)
       .map(first)
 
     let topRows = Signal.combineLatest(
       initialTopFilters,
-      initialSelectedRowWithCategories.map(first)
+      initialSelectedRow
       )
       .map { params, selectedRow -> [SelectableRow] in
         params.map { p in SelectableRow(isSelected: p == selectedRow.params, params: p) }
     }
 
-    let categoryId = self.initialSelectedRowWithCategoriesProperty.signal.skipNil()
-      .map(first)
+    let categoryId = self.initialSelectedRowProperty.signal.skipNil()
       .map { $0.params.category?.rootId }
 
+    let areLoading = MutableProperty(false)
+    self.categoriesAreLoading = areLoading.signal
```

Comment:
i think there is a way to express this without using a mutable property!

you could do a merging of `viewDidLoad.mapConst(true)` and `categoriesEvent.mapConst(false)`. the latter signal emails when it gets a value or error, which means the api request finished!